src/com/toopher/openam/ToopherSecondFactor.java

```diff
@@ -73,154 +68,80 @@ private void debug_message(String message) {
 
     @Override
     public void init(Subject subject, Map sharedState, Map options) {
-        super.init(subject, sharedState, options);
+        if (isUseFirstPassEnabled()) {
+            this.sharedState = sharedState;
+        }
+        if (sharedState == null) {
+            debug_message("sharedState is null!");
+        }
+        if (options == null) {
+            debug_message("options is null");
+        }
+        this.options = options;
```

Comment:
Because we don't do anything more than log when we encounter `null`, I'd vote to always print the `sharedState` and `options` values and trim the conditionals:
```
debug_message("sharedState is " + sharedState);
debug_message("options is " + options);
if (isUseFirstPassEnabled()) {
    this.sharedState = sharedState;
}
this.options = options;
```